cscs-checks/prgenv/affinity_check.py

```diff
@@ -0,0 +1,155 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+class AffinityTestBase(rfm.RegressionTest):
+    def __init__(self, variant):
+        super().__init__()
+        self.valid_systems = ['daint:gpu', 'daint:mc',
+                              'dom:gpu', 'dom:mc']
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.build_system = 'Make'
+        self.build_system.options = ['--directory=affinity', 'MPI=1']
+        self.prebuild_cmd = ['git clone https://github.com/vkarak/affinity']
+        self.executable = 'affinity/affinity'
+        self.variant = variant
+
+        self.maintainers = ['RS', 'VK']
+        self.tags = {'production', 'scs'}
+
+    def setup(self, partition, environ, **job_opts):
+
+        def parse_affinity(x):
+            x_int = [int(xi) for xi in x.split()]
+            return sorted(x_int)
+
+        re_aff_cores = r'CPU affinity: \[\s+(?P<cpu>[\d+\s+]+)\]'
+        self.aff_cores = sn.extractall(
+            re_aff_cores, self.stdout, 'cpu', parse_affinity)
+        self.ref_cores = sn.extractall(
+            re_aff_cores, self.cases[self.variant]['ref_%s' % partition.name],
+            'cpu', parse_affinity)
+        re_aff_thrds = r'^Tag:[^\n\r]*Thread:\s+(?P<thread>\d+)'
+        self.aff_thrds = sn.extractall(re_aff_thrds, self.stdout, 'thread',
+                                       int)
+        self.ref_thrds = sn.extractall(
+            re_aff_thrds, self.cases[self.variant]['ref_%s' % partition.name],
+            'thread', int)
+        re_aff_ranks = r'^Tag:[^\n\r]*Rank:\s+(?P<rank>\d+)[\s+\S+]'
+        self.aff_ranks = sn.extractall(re_aff_ranks, self.stdout, 'rank', int)
+        self.ref_ranks = sn.extractall(
+            re_aff_ranks, self.cases[self.variant]['ref_%s' % partition.name],
+            'rank', int)
+
+        self.use_multithreading = self.cases[self.variant]['multithreading']
+
+        self.sanity_patterns = sn.all([
+            sn.assert_eq(self.aff_thrds, self.ref_thrds),
+            sn.assert_eq(self.aff_ranks, self.ref_ranks),
+            sn.assert_eq(self.aff_cores, self.ref_cores)])
+
+        super().setup(partition, environ, **job_opts)
+
+
+@rfm.parameterized_test(['omp_bind_threads'],
+                        ['omp_bind_threads_nomultithread'],
+                        ['omp_bind_cores'])
+class AffinityOpenMPTest(AffinityTestBase):
```

Comment:
Don't you have to have 
```
self.build_system.options += ['OPENMP=1']
```
for this case? You are testing OpenMP affinity, but the code is not OpenMP enabled.

Reply:
`OPENMP` is 1 by default on the Makefile.